tests/kernel/io_uring.pm

```diff
@@ -0,0 +1,140 @@
+# SUSE's openQA tests
+#
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Executes liburing testing suite
+# Maintainer: Kernel QE <kernel-qa@suse.de>
+# More documentation is at the bottom
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use utils;
+
+sub run {
+    my $self = shift;
+
+    select_serial_terminal;
+
+    my $repository = get_var('LIBURING_REPO', 'https://github.com/axboe/liburing.git');
+    my $timeout = get_var('LIBURING_TIMEOUT', 1800);
+    my $version = get_var('LIBURING_VERSION', '');
+    my $exclude = get_var('LIBURING_EXCLUDE', '');
+    my $issues = get_var('LIBURING_KNOWNISSUES', '');
+    my $pkgs = "git-core";
+    my @lines;
+    my $out;
+
+    # check if liburing2 is installed and eventually install it
+    $out = script_output("rpm -q liburing2");
+    if ($out =~ /not installed/) {
+        $pkgs .= " liburing2";
+    }
+
+    # install dependences
+    zypper_call("in -t pattern devel_basis");
+    zypper_call("in $pkgs");
+
+    # select latest liburing version which is supported by the system
+    if ($version eq '') {
+        $out = script_output("rpm -qi liburing2 | awk '/Version/ {print \$3}'");
+        $version = "liburing-$out";
+    }
+
+    # download and compile tests
+    assert_script_run("git clone --no-single-branch $repository");
+    assert_script_run("git -C liburing checkout $version");
+    assert_script_run("cd liburing");
+    assert_script_run("./configure");
+    assert_script_run("make -C src");
+    assert_script_run("make -C test");
+
+    # run tests executables
+    if ($exclude ne '') {
+        assert_script_run("echo TEST_EXCLUDE=\"$exclude\" > test/config.local");
+
+        record_info(
+            "Exclude",
+            "Excluding tests: $exclude",
+            result => 'softfail'
+        );
+    }
+
+    $out = script_output(
+        "make -C test runtests",
+        timeout => $timeout,
+        proceed_on_failure => 1
+    );
+
+    # search for timed out tests
+    @lines = $out =~ /Tests timed out \(d+\):.*/mg;
+    if (@lines) {
+        record_info(
+            "Timeout",
+            $lines[0],
+            result => 'softfail'
+        );
+    }
+
+    # search for failed tests and known issues
+    @lines = $out =~ /Tests failed \(\d+\):.*/mg;
+    if (@lines) {
+        my @failures = $lines[0] =~ '<(?P<test>[\w-]+\.t)>';
+        my @known_issues;
+
+        if ($issues ne '') {
+            foreach my $issue (split / /, $issues) {
+                if (grep($issue, @failures)) {
+                    push(@known_issues, $issue);
```

Comment:
Storing known issues in a testsuite setting will be a pain to maintain, not to mention that there'll be no info about why it's whitelisted. Using the existing `LTP::Whitelist` is much simpler. The modification needed to use whitelist URL from another variable is fairly small.

Reply:
But ``LTP::Whitelist`` seems to use ``LTP_KNOWN_ISSUES`` which is a specific variable for LTP. I tried to use it, but it doesn't seem to match with external framework.

Reply:
Also it's not possible to use ``LTP::Whitelist`` because liburing testing suite is executed using ``make runtests`` command, which is needed in order to run **only** the tests liburing is supporting. And also, we should rewrite most of the features implemented in ``liburing/test/runtests.sh`` .

At first I just filtered out tests, but that didn't work really well, because we also need ``dmesg`` check after tests executions and to filter out only the tests which are listed inside the Makefile.

Reply:
BTW in pynfs, we just store every info in the code (not optimal, but at least it does not disappear on next openQA postgres db disaster):
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/f4992c4cae76ae0846e38c9ae3dfd4c102833378/tests/nfs/pynfs_result.pm#L29-L50

+1 to modify LTP known issues that can be use for other testsuites. We have started to use quite lot of them that it'd pay the effort.

Reply:
I'm now using ``LTP::Whitelist``, but its implementation needs an adaptation to add custom known issues environment variables, because ``LTP_KNOWN_ISSUES`` and ``LTP_KNOWN_ISSUES_LOCAL`` are LTP specific.